soci/soci_index.go

```diff
@@ -94,6 +112,60 @@ type IndexDescriptorInfo struct {
 	CreatedAt time.Time
 }
 
+// DecodeIndex deserializes a JSON blob in an io.Reader
+// into a SOCI index. The blob can be either an OCI 1.1 Artifact
+// or an OCI 1.0 Manifest
+func DecodeIndex(r io.Reader, index *Index) error {
+	b, err := io.ReadAll(r)
+	if err != nil {
+		return err
+	}
+	return UnmarshalIndex(b, index)
+}
+
+// UnmarshalIndex deserializes a JSON blob in a byte array
+// into a SOCI index. The blob can be either an OCI 1.1 Artifact
+// or an OCI 1.0 Manifest
+func UnmarshalIndex(b []byte, index *Index) error {
```

Comment:
is this public just for testing?

Reply:
I modeled this like the `json` package which has both `Unmarshal` and `Decode` so that it's convenient whether you have a `[]byte` or `io.Reader`. Since we have to be able to read the data twice, I'd prefer to leave this exported to avoid the extra copy if you go through `Decode(bytes.NewReader(b))`